Stm32h7rs flash restore flash size of 64KB

FRASTM opened the pull request:
1. Fix the double definition of the FLASH_OPT_KEY1 and FLASH_OPT_KEY2 for the stm32h7RS serie
2. following the https://github.com/zephyrproject-rtos/zephyr/pull/89675 exact flash size for the stm32H7RS is fixed to 64KB

erwango reviewed (changes requested):
Not compatible with the changes done here: https://github.com/zephyrproject-rtos/zephyr/pull/89675
It seems github has no rebase problem on this but combination of both won't work.

FRASTM committed:
drivers: flash: stm32h7RS flash size is fixed to 64KBytes

Fix the flash size to 64KB for the stm32H7RS series.
Restore the get_size API function for that series.
This is a specificity of this stm32 mcu (RefMan RM0477).

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
rebase on 88b108d201c696667db123d571562f1c9f81f2ac

etienne-lms commented on drivers/flash/flash_stm32h7x.c:
```diff
@@ -836,17 +836,19 @@ static const struct flash_parameters *flash_stm32h7_get_parameters(const struct
 	ARG_UNUSED(dev);
 
+#ifdef CONFIG_SOC_SERIES_STM32H7RSX
+	*size = (uint64_t)0x10000U; /* This serie has only 64K of user flash memory */
```
Could you replace `serie` with `series`.
This comment also applies to the commit message (several `serie` occurrences).

Otherwise, looks all good to me.

etienne-lms commented on drivers/flash/flash_stm32h7x.c:
```diff
@@ -836,17 +836,19 @@ static const struct flash_parameters *flash_stm32h7_get_parameters(const struct
 	ARG_UNUSED(dev);
 
+#ifdef CONFIG_SOC_SERIES_STM32H7RSX
+	*size = (uint64_t)0x10000U; /* The series has only 64K of user flash memory */
```
I missed that: would be better with `KB(64)`, and you can drop the cast.
Not a strong opinion.

etienne-lms reviewed (approved):
LGTM (despite a nitpicking comment)
